projects/igniteui-angular/src/lib/grids/api.service.ts

```diff
@@ -137,31 +137,24 @@ export class GridBaseAPIService <T extends IgxGridBaseDirective & GridType> {
     }
 
     update_cell(cell: IgxCell, value: any) {
-        const data = this.get_all_data(this.grid.transactions.enabled);
-        const index = this.get_row_index_in_data(cell.id.rowID);
-
         cell.editValue = value;
-
         const args = cell.createEditEventArgs();
-
         // TODO: emit onCellEdit after value is updated - issue #7304
         this.grid.onCellEdit.emit(args);
         if (args.cancel) {
             return args;
         }
 
-        // Cast to number after emit
-        // TODO: Clean up this
-        args.newValue = cell.castToNumber(args.newValue);
-
         if (isEqual(args.oldValue, args.newValue)) {
             return args;
         }
 
         this.grid.summaryService.clearSummaryCache(args);
+        const data = this.get_all_data(this.grid.transactions.enabled);
```

Comment:
as getRowData  returns data[index] instead of all data, I will check what refactoring the method to return both data and data[index] will break and may postpone this one

Reply:
you are do not need `data` or the `index` in the method body. What you need is `rowData` in the `this.updateData` call, and this is exactly what `getRowData` will give you :)

Reply:
done